File: src/sys/java/fan/sys/Param.java
//
// Copyright (c) 2006, Brian Frank and Andy Frank
// Licensed under the Academic Free License version 3.0
//
// History:
//   4 Jan 06  Brian Frank  Creation
//
package fan.sys;

/**
 * Param represents one parameter definition of a Func (or Method).
 */
public class Param
  extends FanObj
{

//////////////////////////////////////////////////////////////////////////
// Constructor
//////////////////////////////////////////////////////////////////////////

  public Param(String name, Type type, int mask)
  {
    this.name = name;
    this.type = type;
    this.mask = mask;
  }

//////////////////////////////////////////////////////////////////////////
// Methods
//////////////////////////////////////////////////////////////////////////

  public Type typeof() { return Sys.ParamType; }

  public String name()  { return name; }
  public Type type()   { return type; }
  public boolean hasDefault() { return (mask & HAS_DEFAULT) != 0; }

  public String toStr() { return type + " " + name; }

//////////////////////////////////////////////////////////////////////////
// Fields
//////////////////////////////////////////////////////////////////////////

  public static final int HAS_DEFAULT   = 0x01;  // is a default value provided

  final String name;
  final Type type;
  final int mask;

}

File: src/sys/dotnet/fan/sys/Param.cs
//
// Copyright (c) 2006, Brian Frank and Andy Frank
// Licensed under the Academic Free License version 3.0
//
// History:
//   23 Jul 07  Andy Frank  Creation
//

namespace Fan.Sys
{
  /// <summary>
  /// Param represents one parameter definition of a Func (or Method).
  /// </summary>
  public class Param : FanObj
  {

  //////////////////////////////////////////////////////////////////////////
  // Constructor
  //////////////////////////////////////////////////////////////////////////

    public Param(string name, Type type, int mask)
    {
      this.m_name = name;
      this.m_type = type;
      this.m_mask = mask;
    }

  //////////////////////////////////////////////////////////////////////////
  // Methods
  //////////////////////////////////////////////////////////////////////////

    public override Type @typeof() { return Sys.ParamType; }

    public string name() { return m_name; }
    public Type type() { return m_type; }
    public bool hasDefault() { return (m_mask & HAS_DEFAULT) != 0; }

    public override string toStr() { return m_type + " " + m_name; }

  //////////////////////////////////////////////////////////////////////////
  // Fields
  //////////////////////////////////////////////////////////////////////////

    public static readonly int HAS_DEFAULT = 0x01;  // is a default value provided

    internal readonly string m_name;
    internal readonly Type m_type;
    internal readonly int m_mask;
  }
}